### PR TITLE
POC for an expressions reference page.

### PR DESCRIPTION
**Pull request by SushrutHarness**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \ POC of a new expressions reference page, removing it from docs describing how to use it.
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 1]
```diff
@@ -0,0 +1,627 @@
+---
+title: Harness Expressions Reference
+description: List of default (built-in) Harness expressions.
+sidebar_position: 21
+redirect_from:
+  - /docs/platform/variables-and-expressions/status-type-reference
+---
+
+## Expressions reference
```
Comment (aimurphy) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:9]:
```suggestion

```

File: docs/platform/variables-and-expressions/harness-variables.md [hunk 2]
```diff
@@ -486,626 +484,6 @@ Some important points to note:
 2. The Execution Context is non-editable, meaning you won't be able to add or remove items from it.
 :::
 
-## Expressions reference
```
Comment (aimurphy) [inline docs/platform/variables-and-expressions/harness-variables.md:489]: I would consider leaving this heading.

```
## Expressions reference

For a list of commonly-used built-in expressions, go to the [Expressions reference](./harness-expressions-reference.md).
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 3]
```diff
@@ -0,0 +1,626 @@
...
+* `<+artifacts.primary.dockerConfigJsonSecret>`: If your Kubernetes cluster doesn't have permission to access a private Docker registry, the `values.yaml` or manifest files in the service definition's **Manifests** section must use the `dockerconfigjson` parameter. Then, if you add the Docker image in the service definition's **Artifacts** section, you can reference it with `dockerconfigjson: <+artifact.dockerConfigJsonSecret>`.
+
+For more information and examples, go to [Pull an Image from a Private Registry for Kubernetes](/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/pull-an-image-from-a-private-registry-for-kubernetes) and [Harness Kubernetes services](/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/kubernetes-services.md#pull-an-image-from-a-private-registry).
+
+#### Rollback artifacts
+
+You can use the syntax, `<+rollbackArtifact.ARTIFACT_DEFINITION_ID>` to pull artifact rollback information. For example, use `<+rollbackArtifact.metadata.image>` to pull the metadata of the artifact image used in the last successful deployment.
```
Comment (douglas-j-bothwell) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:448]:
```suggestion
You can use the syntax `<+rollbackArtifact.ARTIFACT_DEFINITION_ID>` to pull artifact rollback information. For example, use `<+rollbackArtifact.metadata.image>` to pull the metadata of the artifact image used in the last successful deployment.
```

**Review by douglas-j-bothwell — APPROVED** [review]
A few optional sugs otherwise LGTM

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 4]
```diff
@@ -0,0 +1,626 @@
+---
+title: Harness Expressions Reference
+description: List of default (built-in) Harness expressions.
+sidebar_position: 21
+redirect_from:
+  - /docs/platform/variables-and-expressions/status-type-reference
+---
+
+
+The following sections describe some Harness expressions. This information is not exhaustive.
```
[the remark this answers is not in the document]
Comment (SushrutHarness) [reply docs/platform/variables-and-expressions/harness-expressions-reference.md:10]: Yes I should link to /docs/platform/variables-and-expressions/harness-variables.md! 

Unfortunately though, there is no other location currently where a user (Or even someone internally) can go to answer the question "Where can I find information about other harness expressions?"

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 5]
```diff
@@ -0,0 +1,626 @@
...
+
+* `<+artifact.metadata.fileName>` and `<+artifact.metadata.url>`: The artifact metadata file name and metadata file URL. Not applicable to all artifact types. If populated, you can find these values in the execution details for the **Service** step, under the **Output** tab. For more information, go to [CD artifact sources](/docs/continuous-delivery/x-platform-cd-features/services/artifact-sources).
+
+   ![](./static/artifact-file-name-variable.png)
+
+#### Artifacts with dockercfg or dockerconfigjson
+
+* `<+artifacts.primary.imagePullSecret>`: If your Kubernetes cluster doesn't have permission to access a private Docker registry, the `values.yaml ` or manifest file in the service definition's **Manifests** section must use the `dockercfg` parameter. Then, if you add the Docker image in the service definition's **Artifacts** section, you can reference it with `dockercfg: <+artifacts.primary.imagePullSecret>`.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:441]:
```suggestion
* `<+artifacts.primary.imagePullSecret>`: If your Kubernetes cluster doesn't have permission to access a private Docker registry, the `values.yaml` file or manifest file in the service definition's **Manifests** section must use the `dockercfg` parameter. Then, if you add the Docker image in the service definition's **Artifacts** section, you can reference it with `dockercfg: <+artifacts.primary.imagePullSecret>`.
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 6]
```diff
@@ -0,0 +1,626 @@
...
+* `<+artifacts.primary.dockerConfigJsonSecret>`: If your Kubernetes cluster doesn't have permission to access a private Docker registry, the `values.yaml` or manifest files in the service definition's **Manifests** section must use the `dockerconfigjson` parameter. Then, if you add the Docker image in the service definition's **Artifacts** section, you can reference it with `dockerconfigjson: <+artifact.dockerConfigJsonSecret>`.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:442]:
```suggestion
* `<+artifacts.primary.dockerConfigJsonSecret>`: If your Kubernetes cluster doesn't have permission to access a private Docker registry, the `values.yaml` file or manifest files in the service definition's **Manifests** section must use the `dockerconfigjson` parameter. Then, if you add the Docker image in the service definition's **Artifacts** section, you can reference it with `dockerconfigjson: <+artifact.dockerConfigJsonSecret>`.
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 7]
```diff
@@ -0,0 +1,626 @@
...
+
+You can also have [rollback artifacts](#rollback-artifacts) and [sidecar artifacts](#sidecar-artifacts).
+
+For a detailed example, go to [Add container images as artifacts for Kubernetes deployments](/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/add-artifacts-for-kubernetes-deployments.md) and [Add a Kubernetes sidecar container](/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/add-a-kubernetes-sidecar-container.md)
+
+</details>
+
+#### Primary artifact names and paths
+
+Use `<+artifacts.primary.image>` or `<+artifacts.primary.imagePath>` in your values YAML file when you want to deploy an artifact you have added to the **Artifacts** section of a CD stage service definition.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:382]:
```suggestion
Use `<+artifacts.primary.image>` or `<+artifacts.primary.imagePath>` in your `values.yaml` file when you want to deploy an artifact you have added to the **Artifacts** section of a CD stage service definition.
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 8]
```diff
@@ -0,0 +1,626 @@
...
+
+For more information and examples, go to [Pull an Image from a Private Registry for Kubernetes](/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/cd-kubernetes-category/pull-an-image-from-a-private-registry-for-kubernetes) and [Harness Kubernetes services](/docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/kubernetes-services.md#pull-an-image-from-a-private-registry).
+
+#### Rollback artifacts
+
+You can use the syntax `<+rollbackArtifact.ARTIFACT_DEFINITION_ID>` to pull artifact rollback information. For example, use `<+rollbackArtifact.metadata.image>` to pull the metadata of the artifact image used in the last successful deployment.
+
+Harness pulls rollback artifact information from last successful deployment. If there's no previous successful deployment, then rollback artifact expressions resolve to `null`.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:450]:
```suggestion
Harness pulls rollback artifact information from the last successful deployment. If there's no previous successful deployment, then rollback artifact expressions resolve to `null`.
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 9]
```diff
@@ -0,0 +1,626 @@
...
+* `<+stage.description>`: The description of the stage.
+* `<+stage.tags>`: The [tags](/docs/platform/references/tags-reference) on the stage. To reference a specific tag, use `<+stage.tags.TAG_NAME>`. To reference tags from a stage outside the stage where you use the expression, use `<+pipeline.stages.STAGE_ID.tags.TAG_NAME>`.
+* `<+stage.identifier>`: The [identifier](../references/entity-identifier-reference.md) of the stage.
+* `<+stage.output.hosts>`: Lists all of the target hosts when deploying to multiple hosts.
+
+   When you are deploying to multiple hosts, such as with an SSH, WinRM, or deployment template stage, you can run the same step on all of the target hosts. To run the step on all hosts, use a [repeat looping strategy](../pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism.md) and identify all the hosts for the stage as the target. For more information and examples go to [Deployment instance expressions](#deployment-instance-expressions) and [Secure Shell (SSH) deployments](/docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng).
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:489]:
```suggestion
   When you are deploying to multiple hosts, such as with an SSH, WinRM, or deployment template stage, you can run the same step on all of the target hosts. To run the step on all hosts, use a [repeat looping strategy](../pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism.md) and identify all the hosts for the stage as the target. For more information and examples, go to [Deployment instance expressions](#deployment-instance-expressions) and [Secure Shell (SSH) deployments](/docs/continuous-delivery/deploy-srv-diff-platforms/traditional/ssh-ng).
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 10]
```diff
@@ -0,0 +1,626 @@
...
+
+`status` refers to the running status of a single node. `currentStatus` and `liveStatus` provide the combined statuses of all running steps within a pipeline or stage. The difference between status types is based on how they handle step failures and if the status of steps running in a matrix or strategy is included in the overall status calculation.
+
+* **Status**: `status` expressions (such as `<+pipeline.stages.STAGE_ID.status>`) refer to the current running status of a single node, such as a pipeline, stage, or step. It provides information about the state of that specific node without considering the status of any parent, child, or sibling nodes. It reports the direct status of the target node.
+* **Current Status**: `currentStatus` expression (such as `<+pipeline.stages.STAGE_ID.currentStatus>`) represent the combined status of all the running steps within a pipeline or stage, except steps generated from [matrix/repeat looping strategies](/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism.md).
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:529]:
```suggestion
* **Current Status**: `currentStatus` expressions (such as `<+pipeline.stages.STAGE_ID.currentStatus>`) represent the combined status of all the running steps within a pipeline or stage, except steps generated from [matrix/repeat looping strategies](/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism.md).
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 11]
```diff
@@ -0,0 +1,626 @@
...
+
+   `currentStatus` uses the statuses of all non-matrix steps to determines the overall status. If *any* non-matrix step fails, regardless of the progress or status of other steps, the `currentStatus` of both the pipeline and the stage resolves as `Failed`. This means that the failure of one step can affects the status of the entire pipeline or stage.
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:531]:
```suggestion
   `currentStatus` uses the statuses of all non-matrix steps to determine the overall status. If *any* non-matrix step fails, regardless of the progress or status of other steps, the `currentStatus` of both the pipeline and the stage resolves as `Failed`. This means that the failure of one step can affect the status of the entire pipeline or stage.
```

File: docs/platform/variables-and-expressions/harness-expressions-reference.md [hunk 12]
```diff
@@ -0,0 +1,626 @@
...
+
+   :::info
+
+   `currentStatus` *ignores* steps generated from matrix/repeat looping strategies. This means that if a pipeline includes a step generated from a matrix, and the matrix step fails while all other steps succeed, then the `currentStatus` is `Success` because `currentStatus` ignores the matrix step.
+
+   :::
+
+* **Live Status**: Like `currentStatus`, `liveStatus` expressions (such as `<+pipeline.stages.stage1.liveStatus>`) also provides the combined status of all the running steps within a pipeline or stage; however it also considers the status of steps generated from [matrix/repeat looping strategies](/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism.md).
```
Comment (brian-f-harness) [inline docs/platform/variables-and-expressions/harness-expressions-reference.md:539]:
```suggestion
* **Live Status**: Like `currentStatus`, `liveStatus` expressions (such as `<+pipeline.stages.stage1.liveStatus>`) also provide the combined status of all the running steps within a pipeline or stage; however, it also considers the status of steps generated from [matrix/repeat looping strategies](/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism.md).
```

**Review by brian-f-harness — DISMISSED** [review]
Left some minor suggestions. Looks good.

**Pull request merged**